### PR TITLE
KAN-52/DPA/-Implement-tags-REST-API

### PR DESCRIPTION
**Pull request by Dpavlovski**

### CODE REVIEW

File: backend/src/main/resources/db/migration/V1_3__add_initial_tags.sql
Comment (HalidOdat) [inline backend/src/main/resources/db/migration/V1_3__add_initial_tags.sql:1]: It's better to have it initialize entries on the java side, we should keep the schema separate from the data.

See this link on how to do that: <https://www.baeldung.com/running-setup-logic-on-startup-in-spring#1-the-postconstruct-annotation>

File: backend/src/main/java/org/latexscribe/LatexScribe/webcontroller/TagController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/tags")
+public class TagController {
+    private final ITagService tagService;
+
+    @GetMapping
```
Comment (HalidOdat) [inline backend/src/main/java/org/latexscribe/LatexScribe/webcontroller/TagController.java]:
```suggestion
    @GetMapping("/")
```

I know `GetMapping` without arguments uses the root, but I would still prefer to have an explicit definition.

File: backend/src/main/java/org/latexscribe/LatexScribe/webcontroller/TagController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/tags")
+public class TagController {
```
Comment (HalidOdat) [inline backend/src/main/java/org/latexscribe/LatexScribe/webcontroller/TagController.java:16]: Nice, but we should probably update the specification on the jira pages

**Review by HalidOdat — COMMENTED** [review]
Looks good just some small adjustments :)

File: backend/src/main/resources/db/migration/V1_3__add_initial_tags.sql
Comment (Dpavlovski) [reply backend/src/main/resources/db/migration/V1_3__add_initial_tags.sql:1]: I don't think this is necessary, we can talk it over on the next meeting.

**Pull request merged**